openai/src/commonMain/kotlin/com/xebia/functional/xef/conversation/llm/openai/OpenAI.kt

```diff
@@ -22,12 +22,32 @@ import kotlin.jvm.JvmField
 import kotlin.jvm.JvmOverloads
 import kotlin.jvm.JvmStatic
 import kotlin.jvm.JvmSynthetic
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.Duration.Companion.seconds
 
 private const val KEY_ENV_VAR = "OPENAI_TOKEN"
 private const val HOST_ENV_VAR = "OPENAI_HOST"
 
-class OpenAI(internal var token: String? = null, internal var host: String? = null) :
-  AutoCloseable, AutoClose by autoClose() {
+class OpenAI(
+  internal var token: String? = null,
+  internal var host: String? = null,
+  internal var timeout: Timeout = Timeout.default()
+) : AutoCloseable, AutoClose by autoClose() {
+
+  class Timeout private constructor(
```

Comment:
Can we relax the `private constructor` to just public? otherwise, a user would never be able to construct a time out without using the `Timeout.default()`. In practice, the time-out policy depends on the use case. The current default timeout is a good policy for a server, but a user-facing client or a different use case may not want to wait minutes for connection or socket timeouts.

Reply:
Totally agree, depends of the use case, so is better remove the private constructor. Thanks for the suggestion! :metal: